### ISSUE TITLE
Support for arm64 platform

### ISSUE THREAD

**Issue by diashenrique** [intersystems/language-server#138]
Hi, 

I'm using the arm64 platform, and when I started the VSCode, the following error message pops up:

[Info  - 8:27:26 AM] Connection to server got closed. Server will restart.
/Users/diashenrique/.vscode-insiders/extensions/intersystems.language-server-1.1.2/server/node_modules/node-gyp-build/index.js:59
  throw new Error('No native build was found for ' + target + '\n    loaded from: ' + dir + '\n')
  ^

Error: No native build was found for platform=darwin arch=arm64 runtime=electron abi=85 uv=1 armv=8 libc=glibc node=12.18.3 electron=11.3.0
    loaded from: /Users/diashenrique/.vscode-insiders/extensions/intersystems.language-server-1.1.2/server/node_modules/ref-napi

    at Function.load.path (/Users/diashenrique/.vscode-insiders/extensions/intersystems.language-server-1.1.2/server/node_modules/node-gyp-build/index.js:59:9)
    at load (/Users/diashenrique/.vscode-insiders/extensions/intersystems.language-server-1.1.2/server/node_modules/node-gyp-build/index.js:21:30)
    at Object.<anonymous> (/Users/diashenrique/.vscode-insiders/extensions/intersystems.language-server-1.1.2/server/node_modules/ref-napi/lib/ref.js:8:53)
    at Module._compile (internal/modules/cjs/loader.js:1199:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1039:32)
    at Module._load (internal/modules/cjs/loader.js:932:14)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12738)
    at Module.require (internal/modules/cjs/loader.js:1079:19)
    at require (internal/modules/cjs/helpers.js:72:18)
[Error - 8:27:27 AM] Connection to server got closed. Server will not be restarted.

BTW, the VSCode IRIS extension is working; I just ignore the message.

**Comment by isc-bsaviano** [comment]
Similar to #32 & #38

**Comment by diashenrique** [comment]
I have looked at issues #32 e #38, but the issues on arm64 it's happening. 
I'm using the language server 1.1.6
Sorry for the questioning, but even with the extension working, I have many drawbacks as No Intellisense or any other feature.
When I said the extension working, I talked about the elementary stuff as compilation, connecting to IRIS, and just it.
Is there anything I can try?

**Comment by isc-bsaviano** [comment]
@diashenrique Do you still get the same error as above? The extension includes compiled C++ libraries. We don't have versions of them for arm64

**Comment by diashenrique** [comment]
@isc-bsaviano Yes, I'm still getting the same error as above.
My MacBook is the M1 version, is there anything that I can help you with?

**Comment by isc-bsaviano** [comment]
@diashenrique Unfortunately you'll just have to disable the LS extension until we build a version of parser libraries that run on arm64. The main extension will provide some basic syntax coloring and intellisense if the LS is disabled or not installed.

**Comment by diashenrique** [comment]
Thanks, I'm looking forward to an arm64 version :)

**Comment by isc-bsaviano** [comment]
@bg-ssh This is still planned, but I don't have an ETA. We're blocked by other teams within InterSystems that have more critical issues on their plate

**Comment by smg1** [comment]
Looking forward to the Arm64 build as well

**Comment by isc-bsaviano** [comment]
Still no updates unfortunately. This is still being explored. I will post any updates when I have them.

**Comment by sethjberkowitz** [comment]
Any updates on Intersystems language server compatibility for Apple Silicon or any work-arounds that others have found?

**Comment by isc-bsaviano** [comment]
@sethjberkowitz This is in active development. We are working towards a solution that will support arm and improve performance for all platforms. I will update this issue when there's a beta vsix for arm users to test.

**Comment by isc-bsaviano** [comment]
Language Server version 2.0.0 was just released and it contains support for arm64 on Mac.

**Comment by mhoritaisj** [comment]
Thanks for this update.

But even with v 2.0.0, the language server crashes if the code contains methods written in Python with the error:

[Info  - 3:02:38 PM] Connection to server got closed. Server will restart.
[Error - 3:02:38 PM] Request textDocument/documentSymbol failed.
Error: Connection got disposed.
	at Object.dispose (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:124428)
	at Object.dispose (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:207144)
	at j.handleConnectionClosed (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:207332)
	at j.handleConnectionClosed (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:284585)
	at t (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:205526)
	at n.invoke (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:126056)
	at r.fire (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:126821)
	at Z (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:112952)
	at n.invoke (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:126056)
	at r.fire (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:126821)
	at d.fireClose (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:134707)
	at ChildProcess.<anonymous> (/Users/horita/.vscode/extensions/intersystems.language-server-2.0.0/client/out/extension.js:2:145705)
	at ChildProcess.emit (node:events:402:35)
	at ChildProcess.emit (node:domain:475:12)
	at maybeClose (node:internal/child_process:1064:16)
	at Socket.<anonymous> (node:internal/child_process:450:11)
	at Socket.emit (node:events:390:28)
	at Socket.emit (node:domain:475:12)
	at Pipe.<anonymous> (node:net:687:12)
[

**Comment by isc-bsaviano** [comment]
@mhoritaisj I'm not seeing this error on my Intel Mac. I will ping an ISC dev with an M1 Mac and have them test this for me.

**Comment by isc-bsaviano** [comment]
@mhoritaisj The ISC dev said that they haven't seen any issues with parsing python using an M1 Mac. Can you paste an example of some code that when included breaks the language server but without parses fine?

**Comment by CraigRegester** [comment]
Yes, working much better! I can now use the Go to Definition and hover over appears to work on everything appropriately now. Anything else I should check? Those are the main components I use and they look fine in both my UCR (Trusted CA Cert) and I4H (Local CA Cert) workspaces.

**Comment by mhoritaisj** [comment]
> @mhoritaisj The ISC dev said that they haven't seen any issues with parsing python using an M1 Mac. Can you paste an example of some code that when included breaks the language server but without parses fine?

**Comment by mhoritaisj** [comment]
> @mhoritaisj The ISC dev said that they haven't seen any issues with parsing python using an M1 Mac. Can you paste an example of some code that when included breaks the language server but without parses fine?

@isc-bsaviano  As I look into this further, it is showing weird behavior.  The following simple code opens perfectly OK, but as I edit it, for example, adding an import statement, the LS crashes. I have not identified the exact trigger that causes the crash.

---
Class User.LSTest Extends %RegisteredObject
{
Method doSomething(s As %String) As %String
{
    //Do ^User.Rtn
    Set a = "あ"
    Quit "Received: "_s
}

Method doSomethingInPython(s As %String) As %String [ Language = python ]
{
    a = "Python"
}
}
----

**Comment by isc-bsaviano** [comment]
@mhoritaisj I was able to reproduce your problem on my Intel Mac. I will open a new issue for it and explain what I'm seeing in more detail.

**Comment by mvhenderson** [comment]
@isc-bsaviano two questions:

1. Does v2.x need to be manually downloaded/installed?   In VSCode extensions pane highest version is v1.2.9
2. Is Windows 11 ARM supported? It is not according to the [overview](https://marketplace.visualstudio.com/items?itemName=intersystems.language-server&ssr=false#overview) 

I'm running Win11 ARM VM guest on M2 mac host.

**Comment by isc-bsaviano** [comment]
@mvhenderson Windows 11 ARM is not supported, which is why you're not seeing v2 in the extensions pane. v1 was listed as "Universal" support because it predated VS Code Marketplace's support for platform-specific versions. If you want support for Windows ARM you can create a new issue and I can look into the feasibility but since nothing else at InterSystems (AFAIK) supports Windows ARM I can't promise that it will happen. You can try manually downloading Windows Intel v2 from the marketplace's website and try that but idk if it will work.
